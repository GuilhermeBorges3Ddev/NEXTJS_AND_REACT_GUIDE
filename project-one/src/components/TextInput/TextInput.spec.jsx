// @ts-nocheck
import React from 'react';
import { screen, render } from '@testing-library/react';
import { TextInput } from '.';
import userEvent from '@testing-library/user-event';

describe('<TextInput />', () => {
  it('should have a value of the searchValue prop', () => {
    render(<TextInput handleChange={() => {}} searchValue={'Testing...'} />);
    const input = screen.getByPlaceholderText(/Type your search.../i);
    expect(input.value).toBe('Testing...');
  });
  it('should call handleChange function for each key pressed', () => {
    const fn = jest.fn();
    const value = 'New value';
    render(<TextInput handleChange={fn} searchValue={value} />);
    const input = screen.getByPlaceholderText(/Type your search.../i);
    userEvent.type(input, value);
    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });
  it('should match with the snapshot for TextInput already tested', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} searchValue="" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
