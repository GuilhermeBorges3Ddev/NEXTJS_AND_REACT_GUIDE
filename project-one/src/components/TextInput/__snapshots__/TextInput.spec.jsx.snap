// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match with the snapshot for TextInput already tested 1`] = `
<input
  class="text-input"
  placeholder="Type your search..."
  type="search"
  value=""
/>
`;
